### PR TITLE
[#1337] Make goto variable definition respect list comprehension scopes

### LINKED ISSUE #1337: Go-to-definition for variables behaves incorrectly
Given the following examples, where the `~` represents the cursor location:

```
one() ->
    Var = 1,
    [V~ar || Var <- [1, 2, 3]].
     
two() ->
    [ Var || Var <- [1, 2, 3] ],
    [ V~ar || Var <- [4, 5, 6] ].
```

Go-to-definition jumps to the locations indicated by `*`:

```
one() ->
    *Var = 1,
    [Var || Var <- [1, 2, 3]].
     
two() ->
    [ *Var || Var <- [1, 2, 3] ],
    [ Var || Var <- [4, 5, 6] ].
```

Which is surprising for the user. Instead, they should jump to:

```
one() ->
    Var = 1,
    [Var || *Var <- [1, 2, 3]].
     
two() ->
    [ Var || Var <- [1, 2, 3] ],
    [ Var || *Var <- [4, 5, 6] ].
```

### PR DESCRIPTION
**Pull request by plux**
### Description

Make goto variable definition respect list comprehension scopes.

Fixes #1337 .

### CODE REVIEW

File: apps/els_lsp/test/els_rename_SUITE.erl [hunk 1]
```diff
@@ -277,6 +278,99 @@ rename_variable(Config) ->
...
+    Result6 = rename(Uri, 18, 21, NewName),
+    Expected6 = #{
+        changes => #{
+            UriAtom => [
+                change(NewName, {17, 12}, {17, 16}),
+                change(NewName, {18, 21}, {18, 25})
+            ]
+        }
+    },
+    %% four: FIXME: Bug, shouldn't rename Var inside second LC
```
Comment (plux) [inline apps/els_lsp/test/els_rename_SUITE.erl:354]: Still at least one corner case to fix, leaving that for the future.
